content/docs/for-buildpack-authors/how-to/write-buildpacks/specify-env.md

```diff
@@ -3,15 +3,160 @@ title="Specify the environment"
+
+```text
+
+layers/
+└── some-buildpack-id
+    ├── some-build-layer
+    │   ├── bin
+    │   │   └── some-binary
+    │   ├── env
+    │   │   └── SOME_VAR # contents foo
+    │   ├── env.build
+    │   │   └── SOME_VAR # contents bar
+    │   └── lib
+    │       ├── some-shared-library
+    │       └── some-static-library
+    └── some-build-layer.toml
+
+```
+
+| Env Variable                               | Layer Path   | Contents         | Build | Launch |
+|--------------------------------------------|--------------|------------------|-------|--------|
+| `PATH`                                     | `/bin`       | binaries         | [x]   | [x]    |
+| `LD_LIBRARY_PATH`                          | `/lib`       | shared libraries | [x]   | [x]    |
+| `LIBRARY_PATH`                             | `/lib`       | static libraries | [x]   |        |
+| `CPATH`                                    | `/include`   | header files     | [x]   |        |
+| `PKG_CONFIG_PATH`                          | `/pkgconfig` | pc files         | [x]   |        |
+
+* To add a directory to the $PATH variable for buildpacks that follow, we need to create a `<layers>/<layer>/bin` directory and add any executables we want to make available within that directory. The `lifecycle` will automatically add the new `bin` directory to the `PATH` for buildpacks that follow.
+* Binaries, such as `some-binary`, are added to `PATH` for buildpacks that follow
+* Shared libraries, such as `Some-shared-library` and `some-static-library`, are added to `LD_LIBRARY_PATH` and `LIBRARY_PATH`
+* If `env/SOME_VAR` and `env.build/SOME_VAR` have a conflict, a given procedure applies to figure out who "wins" when applying the environment [modification rules](#environment-variable-modification-rules).
+* `User-defined` variables are then applied; meaning it's possible to override `buildpack-defined` variables.
+* Finally, `platform-defined` variables are applied, which eventually override any previous values.
+
+The above is only relevant for `build` because the layer `some-build-layer` is a build layer, i.e., `some-build-layer.toml` has
+
+```yaml
+
+[types]
+build = true
+
+```
+
+>Note that the platform SHOULD NOT assume any other `base-image` provided environment variables are inherited by the buildpack.
+
+### Build Environment Variables
+
+During the `build` phase, buildpacks MAY write environment variable files to `<layers>/<layer>/env/`, `<layers>/<layer>/env.build/`, and `<layers>/<layer>/env.launch/` directories.
+
+For each `<layers>/<layer>/` designated as a build layer, for each file written to `<layers>/<layer>/env/` or `<layers>/<layer>/env.build/` by `/bin/build`, the `lifecycle` MUST modify an environment variable in subsequent executions of `/bin/build` according to the [environment variable modification rules](https://github.com/buildpacks/spec/blob/main/buildpack.md#environment-variable-modification-rules).
+
+### Process and Runtime Environment Variables
```

Comment:
```suggestion
### Preparing the environment at runtime
```

Reply:
It's probably worth noting here that it's still the "lifecycle" or rather a CNB component that sets up the environment at runtime, but it's the minimal piece of the lifecycle known as the launcher binary, not the lifecycle binary. This can be found inside the app image at `/cnb/lifecycle/launcher` and is the entrypoint for any CNB-built image.